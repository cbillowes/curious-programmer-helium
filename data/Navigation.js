module.exports = [
  {
    url: "/",
    name: "Blog"
  },
  {
    url: "/about",
    name: "About"
  },
  {
    url: "/archives",
    name: "Archives"
  },
  {
    url: "/tags",
    name: "Tags"
  },
  {
    url: "/credits",
    name: "Credits"
  },
  {
    url: "/privacy-policy",
    name: "Privacy Policy"
  },
  {
    url: "/license",
    name: "License"
  }
]
