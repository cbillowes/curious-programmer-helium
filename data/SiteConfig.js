module.exports = {
  siteTitle: "Curious Programmer", // Site title.
  siteTitleShort: "{ curious }", // Short site title for homescreen (PWA). Preferably should be under 12 characters to prevent truncation.
  siteTitleAlt: "Curious Programmer", // Alternative site title for SEO.
  siteLogo: "/images/logos/logo.png", // Logo used for SEO and manifest.
  siteUrl: "https://curiousprogrammer.io", // Domain of your website without pathPrefix.
  pathPrefix: "/", // Prefixes all links. For cases when deployed to example.github.io/gatsby-advanced-starter/.
  blogPostDir: "posts", // The name of directory that contains your posts.
  defaultOgImage: "/images/og/default-og-image.png",
  siteDescription: "A curious place for a curious mind.", // Website description used for RSS feeds/meta description tag.
  siteRss: "/feed.xml", // Path to the RSS file.
  siteFBAppID: "", // FB Application ID for using app insights
  googleAnalyticsID: "UA-77127562-2", // GA tracking ID.
  disqusShortname: "curious-programmer", // Disqus shortname.
  postDefaultCategoryID: "Tech", // Default category for posts.
  dateFromFormat: "YYYY-MM-DD", // Date format used in the frontmatter.
  dateFormat: "DD/MM/YYYY", // Date format for display.
  userName: "Clarice Bouwer", // Username to display in the author segment.
  userTwitter: "cbillowes", // Optionally renders "Follow Me" in the UserInfo segment.
  userLocation: "Johannesburg, South Africa", // User location to display in the author segment.
  userAvatar: "https://en.gravatar.com/userimage/34549463/acc6b8778c7b14437b7590078012302e.png?size", // User avatar to display in the author segment.
  userDescription:
    "I am a curious software developer eager to share what I learn with you.", // User description to display in the author segment.
  // Links to social profiles/projects you want to display in the author segment/navigation bar.
  userLinks: [
    {
      label: "GitHub",
      url: "https://github.com/Vagr9K/gatsby-advanced-starter",
      iconClassName: "fa fa-github"
    },
    {
      label: "Twitter",
      url: "https://twitter.com/Vagr9K",
      iconClassName: "fa fa-twitter"
    },
    {
      label: "Email",
      url: "mailto:vagr9k@gmail.com",
      iconClassName: "fa fa-envelope"
    }
  ],
  themeColor: "#c62828", // Used for setting manifest and progress theme colors.
  backgroundColor: "#e0e0e0" // Used for setting manifest background color.
};
