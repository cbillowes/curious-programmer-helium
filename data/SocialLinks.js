module.exports = {
  footer: [{
    name: "Github",
    href: "https://github.com/cbillowes",
    icon: "github"
  }, {
    name: "Twitter",
    href: "https://twitter.com/cbillowes",
    icon: "twitter"
  }, {
    name: "LinkedIn",
    href: "https://www.linkedin.com/in/cbouwer/",
    icon: "linkedin"
  }],
  contact: [{
    name: "email",
    href: "mailto:curiousprogrammer.io@gmail.com",
    icon: "email"
  }, {
    name: "RSS",
    href: "/rss.xml",
    icon: "rss"
  }, {
    name: "Twitter",
    href: "https://twitter.com/cbillowes",
    icon: "twitter"
  }, {
    name: "StackOverflow",
    href: "https://stackoverflow.com/users/849986/cbillowes",
    icon: "stack-overflow"
  }, {
    name: "Github",
    href: "https://github.com/cbillowes",
    icon: "github"
  }, {
    name: "LinkedIn",
    href: "https://www.linkedin.com/in/cbouwer/",
    icon: "linkedin"
  }]
}