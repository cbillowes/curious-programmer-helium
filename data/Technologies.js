module.exports = [
  {
    name: "Clojure",
    current: true,
    start: new Date(2018, 7, 1)
  },
  {
    name: "Datomic",
    current: true,
    start: new Date(2018, 7, 1)
  },
  {
    name: "C#",
    xp: 10
  },
  {
    name: "React",
    current: true,
    start: new Date(2018, 6, 26)
  },
  {
    name: "Git",
    current: true,
    start: new Date(2014, 4, 1)
  },
  {
    name: "Gatsby",
    current: true,
    start: new Date(2018, 6, 26)
  },
  {
    name: "T-SQL",
    xp: 11
  },
  {
    name: "JavaScript",
    current: true,
    start: new Date(2006, 1, 1)
  },
  {
    name: "jQuery",
    current: true,
    start: new Date(2009, 1, 1)
  },
  {
    name: "VB.NET",
    xp: 3
  },
  {
    name: "Jenkins",
    xp: 5
  },
  {
    name: "MVC"
  },
  {
    name: "Grunt"
  },
  {
    name: "Solr",
    current: true,
    start: new Date(2018, 12, 1)
  },
  {
    name: "WebAPI"
  },
  {
    name: "Handlebars"
  },
  {
    name: "LINQ"
  },
  {
    name: "Reagent",
    current: true,
    start: new Date(2018, 7, 1)
  },
  {
    name: "Web Forms"
  },
  {
    name: "Entity Framework"
  },
  {
    name: "Bootstrap"
  },
  {
    name: "Semantic UI",
    current: true,
    start: new Date(2018, 7, 1)
  },
  {
    name: "CSS/3",
    current: true,
    start: new Date(2004, 1, 1)
  },
  {
    name: "SASS/LESS",
    current: true
  },
  {
    name: "HTML/5",
    current: true,
    start: new Date(2004, 1, 1)
  },
  {
    name: "ClojureScript",
    current: true,
    start: new Date(2018, 7, 1)
  },
  {
    name: "Agile",
    xp: 5
  },
  {
    name: "Clickatel Integration",
    xp: 1
  },
  {
    name: "XMPie Integration",
    xp: 2
  },
  {
    name: "VCS Integration",
    xp: 1
  },
  {
    name: "Nedbank iVeri Lite Integration",
    xp: 3
  },
  {
    name: "Mashreq Payment Gateway Integration",
    xp: 1
  },
  {
    name: "Microsoft SQL Server",
    xp: 11
  }
]
