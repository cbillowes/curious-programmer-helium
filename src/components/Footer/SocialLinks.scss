.social-links-footer {
    float: right;
    margin: 0;

    .link {
        float: right;
        list-style: none;
        margin: 0 0.5em;

        a {
            display: inline-block;
            width: 2.5em;
            height: 2.5em;
            background-size: cover;
        }

        .github {
            background-image: url(data:image/svg+xml;utf8;base64,PD94bWwgdmVyc2lvbj0iMS4wIj8+CjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2aWV3Qm94PSIwIDAgNTEyIDUxMiIgd2lkdGg9IjUxMnB4IiBoZWlnaHQ9IjUxMnB4Ij48cGF0aCBkPSJtNTEyIDI1N2MwIDEyMC04NC4xMDE1NjIgMjIwLjUtMTk2IDI0Ny41bC0zMC42MDE1NjItOTcuMTk5MjE5aC01OC43OTY4NzZsLTI5LjYwMTU2MiA5Ny4xOTkyMTljLTExMS44OTg0MzgtMjctMTk3LTEyNy41LTE5Ny0yNDcuNSAwLTE0MC42OTkyMTkgMTE1LjMwMDc4MS0yNTcgMjU2LTI1N3MyNTYgMTE2LjMwMDc4MSAyNTYgMjU3em0wIDAiIGZpbGw9IiMzODQ5NDkiLz48cGF0aCBkPSJtNTEyIDI1N2MwIDEyMC04NC4xMDE1NjIgMjIwLjUtMTk2IDI0Ny41bC0zMC42MDE1NjItOTcuMTk5MjE5aC0yOS4zOTg0Mzh2LTQwNy4zMDA3ODFjMTQwLjY5OTIxOSAwIDI1NiAxMTYuMzAwNzgxIDI1NiAyNTd6bTAgMCIgZmlsbD0iIzI5MzkzOSIvPjxwYXRoIGQ9Im0xODEuMjc3MzQ0IDQzMC4wNTg1OTRjLTYuMDc4MTI1IDAtMTIuMDExNzE5LS44NjcxODgtMTcuODI4MTI1LTIuNTc4MTI1LTE1LjEyODkwNy00LjQ2ODc1LTI3LjQyMTg3NS0xNC41NDY4NzUtMzYuNTQ2ODc1LTI5LjkxNDA2My00LjE2MDE1Ni03LjAxNTYyNS04LjQ5NjA5NC0xMS44Nzg5MDYtMTMuNjA1NDY5LTE1LjMwODU5NC01LjAyNzM0NC0zLjM4MjgxMi05LjAzOTA2My00LjY3MTg3NC0xMy4yNzM0MzctNC4zNjMyODFsLTIuNjM2NzE5LTI5Ljg4MjgxMmMxMS4xMTcxODctLjk1MzEyNSAyMS43NTM5MDYgMi4wNjI1IDMyLjU5Mzc1IDkuMzE2NDA2IDguODMyMDMxIDUuOTAyMzQ0IDE2LjI1NzgxMiAxNC4wNjI1IDIyLjcxODc1IDI0LjkxNDA2MyA1LjMwNDY4NyA4LjkyMTg3NCAxMS40MTAxNTYgMTQuMTUyMzQzIDE5LjI1IDE2LjQ2ODc1IDguODA0Njg3IDIuNTg5ODQzIDE3Ljk0MTQwNiAxLjUwNzgxMiAyOS42MzI4MTItMy40NzI2NTdsMTEuODA4NTk0IDI3LjU2NjQwN2MtMTEuMjk2ODc1IDQuODM1OTM3LTIxLjkyOTY4NyA3LjI1MzkwNi0zMi4xMTMyODEgNy4yNTM5MDZ6bTAgMCIgZmlsbD0iI2VjZWNmMSIvPjxwYXRoIGQ9Im00MDAuOTAyMzQ0IDI4Ny4zMDA3ODFjLTEwLjUwMzkwNiAyNy44OTg0MzgtMzYuOTAyMzQ0IDYzLjMwMDc4MS0xMDMuODAwNzgyIDczLjE5OTIxOSA4LjY5OTIxOSAxMi44OTg0MzggMTkuMTk5MjE5IDE5LjgwMDc4MSAxOC44OTg0MzggNDYuODAwNzgxdjk3LjE5OTIxOWMtMTkuMTk5MjE5IDQuODAwNzgxLTM5LjMwMDc4MSA3LjUtNjAgNy41cy0zOS44MDA3ODEtMi42OTkyMTktNTktNy41di05OC40MDIzNDRjMC0yNi42OTkyMTggMTAuMTAxNTYyLTM0LjE5OTIxOCAxNy44OTg0MzgtNDUuNTk3NjU2LTY2Ljg5ODQzOC05LjkwMjM0NC05My4yOTY4NzYtNDUuMzAwNzgxLTEwMy44MDA3ODItNzMuMTk5MjE5LTE0LjA5NzY1Ni0zNy4yMDMxMjUtNi41OTc2NTYtODMuNDAyMzQzIDE4LjAwMzkwNi0xMTIuODAwNzgxLjU5NzY1Ny0uNjAxNTYyIDEuNS0yLjEwMTU2MiAxLjE5OTIxOS0zLTExLjQwMjM0My0zNC4xOTkyMTkgMi4zOTg0MzgtNjIuNjk5MjE5IDMtNjUuNjk5MjE5IDEyLjg5ODQzOCAzLjg5ODQzOCAxNS0zLjkwMjM0MyA1Ni42OTkyMTkgMjEuNTk3NjU3bDcuMTk5MjE5IDQuMjAzMTI0YzMgMS43OTY4NzYgMi4xMDE1NjIuNTk3NjU3IDUuMTAxNTYyLjU5NzY1NyAxNy4zOTg0MzgtNC44MDA3ODEgMzUuNjk5MjE5LTcuNSA1My42OTkyMTktNy41IDE4LjMwMDc4MSAwIDM2LjMwMDc4MSAyLjY5OTIxOSA1NC41OTc2NTYgNy41bDIuMTAxNTYzLjMwMDc4MXMuNTk3NjU2IDAgMi4xMDE1NjItLjg5ODQzOGM1MS44OTg0MzgtMzEuNTAzOTA2IDUwLjA5NzY1Ny0yMS4zMDA3ODEgNjQuMTk1MzEzLTI1LjgwMDc4MS4zMDA3ODEgMyAxNC4xMDE1NjIgMzEuNzk2ODc1IDIuNzAzMTI1IDY1LjY5OTIxOS0xLjUgNC41IDQ1IDQ3LjA5NzY1NiAxOS4yMDMxMjUgMTE1LjgwMDc4MXptMCAwIiBmaWxsPSIjZWNlY2YxIi8+PHBhdGggZD0ibTQwMC45MDIzNDQgMjg3LjMwMDc4MWMtMTAuNTAzOTA2IDI3Ljg5ODQzOC0zNi45MDIzNDQgNjMuMzAwNzgxLTEwMy44MDA3ODIgNzMuMTk5MjE5IDguNjk5MjE5IDEyLjg5ODQzOCAxOS4xOTkyMTkgMTkuODAwNzgxIDE4Ljg5ODQzOCA0Ni44MDA3ODF2OTcuMTk5MjE5Yy0xOS4xOTkyMTkgNC44MDA3ODEtMzkuMzAwNzgxIDcuNS02MCA3LjV2LTM4Ny4zMDA3ODFjMTguMzAwNzgxIDAgMzYuMzAwNzgxIDIuNjk5MjE5IDU0LjYwMTU2MiA3LjVsMi4wOTc2NTcuMzAwNzgxcy42MDE1NjIgMCAyLjEwMTU2Mi0uODk4NDM4YzUxLjg5ODQzOC0zMS41MDM5MDYgNTAuMDk3NjU3LTIxLjMwMDc4MSA2NC4xOTkyMTktMjUuODAwNzgxLjMwMDc4MSAzIDE0LjEwMTU2MiAzMS43OTY4NzUgMi42OTkyMTkgNjUuNjk5MjE5LTEuNSA0LjUgNDUgNDcuMDk3NjU2IDE5LjIwMzEyNSAxMTUuODAwNzgxem0wIDAiIGZpbGw9IiNlMmUyZTciLz48L3N2Zz4K);
        }

        .twitter {
            background-image: url(data:image/svg+xml;utf8;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pgo8IS0tIEdlbmVyYXRvcjogQWRvYmUgSWxsdXN0cmF0b3IgMTkuMC4wLCBTVkcgRXhwb3J0IFBsdWctSW4gLiBTVkcgVmVyc2lvbjogNi4wMCBCdWlsZCAwKSAgLS0+CjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0iTGF5ZXJfMSIgeD0iMHB4IiB5PSIwcHgiIHZpZXdCb3g9IjAgMCA1MTIgNTEyIiBzdHlsZT0iZW5hYmxlLWJhY2tncm91bmQ6bmV3IDAgMCA1MTIgNTEyOyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSIgd2lkdGg9IjUxMnB4IiBoZWlnaHQ9IjUxMnB4Ij4KPGNpcmNsZSBzdHlsZT0iZmlsbDojNjVBMkQ5OyIgY3g9IjI1NiIgY3k9IjI1NiIgcj0iMjU2Ii8+CjxwYXRoIHN0eWxlPSJmaWxsOiMzQTdDQTU7IiBkPSJNMzkzLjAxNCwxMzkuMzI2Yy0yNi43MDMsMjMuMTY5LTUzLjI1Myw0My40NzUtNzQuOTU0LDcxLjg1MiAgYy01My4zODEsNjQuMzcyLTExOC42MTMsMTU1LjctMjA3LjM4NiwxNDIuMDg2bDE1OC42MSwxNTguMzk2YzEzNC40NTYtNi44NzMsMjQxLjQ5Ny0xMTcuNDkzLDI0Mi42ODYtMjUzLjM3NkwzOTMuMDE0LDEzOS4zMjZ6Ii8+CjxwYXRoIHN0eWxlPSJmaWxsOiNGRkZGRkY7IiBkPSJNMzk3Ljg3MiwxNjIuNDcxYy02LjUxMywyLjg4OS0xMy4yNzEsNS4xNjctMjAuMjA4LDYuODE1YzcuNjQ0LTcuMjYxLDEzLjM5LTE2LjM0NiwxNi42MzEtMjYuNDg0ICBjMC45MjYtMi44OTMtMi4yMTktNS4zOTgtNC44MzItMy44NDhjLTkuNjUsNS43MjUtMjAuMDQ0LDEwLjAxNi0zMC44OTQsMTIuNzYyYy0wLjYyOCwwLjE2LTEuMjc2LDAuMjQtMS45MjksMC4yNCAgYy0xLjk3OSwwLTMuODk2LTAuNzMzLTUuNDExLTIuMDY1Yy0xMS41NDItMTAuMTc0LTI2LjM5LTE1Ljc3Ny00MS44MDUtMTUuNzc3Yy02LjY3MiwwLTEzLjQwNSwxLjA0LTIwLjAxNiwzLjA5MSAgYy0yMC40ODcsNi4zNTMtMzYuMjk1LDIzLjI1NC00MS4yNTcsNDQuMTAzYy0xLjg2LDcuODE4LTIuMzYyLDE1LjY0OC0xLjQ5NiwyMy4yNjRjMC4wOTcsMC44NzYtMC4zMTQsMS40ODYtMC41NjksMS43NzIgIGMtMC40NSwwLjUwMi0xLjA4NCwwLjc5MS0xLjc0NSwwLjc5MWMtMC4wNzIsMC0wLjE1LTAuMDAzLTAuMjI0LTAuMDFjLTQ0Ljg0Ni00LjE2OC04NS4yODctMjUuNzcyLTExMy44NjktNjAuODM3ICBjLTEuNDU1LTEuNzg5LTQuMjUzLTEuNTY5LTUuNDE1LDAuNDIyYy01LjU5Niw5LjYwNi04LjU1NCwyMC41ODktOC41NTQsMzEuNzY2YzAsMTcuMTI3LDYuODg0LDMzLjI3LDE4LjgzNyw0NS4wMzkgIGMtNS4wMjctMS4xOTMtOS44OTMtMy4wNy0xNC40MTQtNS41ODJjLTIuMTg4LTEuMjE0LTQuODc3LDAuMzUtNC45MDgsMi44NTFjLTAuMzEsMjUuNDQ1LDE0LjU4OCw0OC4wODcsMzYuOTA1LDU4LjI4MiAgYy0wLjQ1LDAuMDEtMC45LDAuMDE0LTEuMzUsMC4wMTRjLTMuNTM3LDAtNy4xMjEtMC4zMzgtMTAuNjQ1LTEuMDE1Yy0yLjQ2My0wLjQ2Ny00LjUzMiwxLjg2Ny0zLjc2OCw0LjI1MyAgYzcuMjQ2LDIyLjYxOCwyNi43MTcsMzkuMjg4LDUwLjAyMSw0My4wN2MtMTkuMzM5LDEyLjk4My00MS44NjMsMTkuODMtNjUuMzAyLDE5LjgzbC03LjMwNi0wLjAwM2MtMi4yNTUsMC00LjE2LDEuNDY5LTQuNzMsMy42NSAgYy0wLjU2NSwyLjE0NSwwLjQ3NCw0LjQxMywyLjM5Niw1LjUzYzI2LjQxMiwxNS4zNzIsNTYuNTQxLDIzLjQ5NSw4Ny4xMzgsMjMuNDk1YzI2Ljc4NCwwLDUxLjgzOC01LjMxMyw3NC40NjYtMTUuNzk4ICBjMjAuNzQ1LTkuNjA5LDM5LjA3Ni0yMy4zNDUsNTQuNDg2LTQwLjgyN2MxNC4zNTctMTYuMjg2LDI1LjU4MS0zNS4wODUsMzMuMzY1LTU1Ljg3OWM3LjQxOC0xOS44MTYsMTEuMzQtNDAuOTY3LDExLjM0LTYxLjE1NCAgdi0wLjk2NGMwLTMuMjQxLDEuNDY1LTYuMjkxLDQuMDI0LTguMzdjOS43MDYtNy44ODIsMTguMTYtMTcuMTU4LDI1LjEyMi0yNy41NzJDNDAzLjc5NiwxNjQuNTc4LDQwMC44OTYsMTYxLjEzLDM5Ny44NzIsMTYyLjQ3MSAgTDM5Ny44NzIsMTYyLjQ3MXoiLz4KPHBhdGggc3R5bGU9ImZpbGw6I0QxRDFEMTsiIGQ9Ik0zOTcuODcyLDE2Mi40NzFjLTYuNTE1LDIuODg5LTEzLjI3MSw1LjE2Ny0yMC4yMDgsNi44MTVjNy42NDQtNy4yNjEsMTMuMzktMTYuMzQ2LDE2LjYzMi0yNi40ODQgIGMwLjkyNi0yLjg5My0yLjIxOS01LjM5OC00LjgzMi0zLjg0OGMtOS42NSw1LjcyNS0yMC4wNDQsMTAuMDE2LTMwLjg5NCwxMi43NjJjLTAuNjI4LDAuMTYtMS4yNzYsMC4yNC0xLjkyOSwwLjI0ICBjLTEuOTc5LDAtMy44OTYtMC43MzMtNS40MTEtMi4wNjVjLTExLjU0Mi0xMC4xNzQtMjYuMzktMTUuNzc3LTQxLjgwNS0xNS43NzdjLTYuNjcxLDAtMTMuNDA1LDEuMDQtMjAuMDE2LDMuMDkxICBjLTE0LjMyMiw0LjQ0MS0yNi4zNDMsMTQuMDQ4LTMzLjk4NSwyNi41NDZ2MjA1LjQ3N2M2LjIyMi0yLjAyOSwxMi4yOTMtNC40MDMsMTguMTk4LTcuMTM5ICBjMjAuNzQ1LTkuNjA5LDM5LjA3Ni0yMy4zNDUsNTQuNDg2LTQwLjgyN2MxNC4zNTctMTYuMjg3LDI1LjU4MS0zNS4wODUsMzMuMzY1LTU1Ljg3OWM3LjQxOC0xOS44MTYsMTEuMzQtNDAuOTY3LDExLjM0LTYxLjE1NCAgdi0wLjk2NGMwLTMuMjQxLDEuNDY1LTYuMjkxLDQuMDI0LTguMzdjOS43MDYtNy44ODIsMTguMTYtMTcuMTU4LDI1LjEyMi0yNy41NzJDNDAzLjc5NiwxNjQuNTc4LDQwMC44OTYsMTYxLjEzLDM5Ny44NzIsMTYyLjQ3MXoiLz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPC9zdmc+Cg==);
        }

        .linkedin {
            background-image: url(data:image/svg+xml;utf8;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pgo8IS0tIEdlbmVyYXRvcjogQWRvYmUgSWxsdXN0cmF0b3IgMTkuMC4wLCBTVkcgRXhwb3J0IFBsdWctSW4gLiBTVkcgVmVyc2lvbjogNi4wMCBCdWlsZCAwKSAgLS0+CjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0iTGF5ZXJfMSIgeD0iMHB4IiB5PSIwcHgiIHZpZXdCb3g9IjAgMCA1MTIuMDAyIDUxMi4wMDIiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDUxMi4wMDIgNTEyLjAwMjsiIHhtbDpzcGFjZT0icHJlc2VydmUiIHdpZHRoPSI1MTJweCIgaGVpZ2h0PSI1MTJweCI+CjxjaXJjbGUgc3R5bGU9ImZpbGw6IzQzNzZCMTsiIGN4PSIyNTYuMDAxIiBjeT0iMjU2IiByPSIyNTYiLz4KPHBhdGggc3R5bGU9ImZpbGw6IzMzNjY5OTsiIGQ9Ik01MTEuMDE2LDIzMy42MzRMMzk0LjcyOSwxMTcuMTk4Yy03MS4wMzUsOS4yMzctMTQ1LjI3LDcuODU5LTIxNi4zMDUsOS41MTEgIGMtMjkuMzA2LDAtNTguNjEzLDEzLjc5MS01My40NDEsNDYuNTQ1YzEwLjM0Myw2NS41MDgsMTUuNTE1LDEyOS4yOTMsMy40NDgsMTk2LjUyNWMtMS45LDguNTQ1LTUuODk3LDE3LjA4OS0xMS4xMTcsMjUuMDU5ICBMMjMzLjQ4LDUxMS4wMDVjNy40MjEsMC42NDYsMTQuOTI5LDAuOTk2LDIyLjUxOSwwLjk5NmMxNDEuMzg0LDAsMjU2LTExNC42MTYsMjU2LTI1NiAgQzUxMi4wMDEsMjQ4LjQ2MSw1MTEuNjU2LDI0MS4wMDUsNTExLjAxNiwyMzMuNjM0eiIvPgo8cGF0aCBzdHlsZT0iZmlsbDojRkZGRkZGOyIgZD0iTTM3Ni4wMjMsMTA5LjQ2OEgxMzUuOTc5Yy0xNC42NDMsMC0yNi41MSwxMS44NjctMjYuNTEsMjYuNTFWMzc2LjAyICBjMCwxNC42MzksMTEuODY3LDI2LjUxLDI2LjUxLDI2LjUxaDI0MC4wNDJjMTQuNjM5LDAsMjYuNTEtMTEuODcxLDI2LjUxLTI2LjUxVjEzNS45NzggIEM0MDIuNTMzLDEyMS4zMzUsMzkwLjY2MiwxMDkuNDY4LDM3Ni4wMjMsMTA5LjQ2OEwzNzYuMDIzLDEwOS40Njh6IE0yMDAuMTU1LDM2Mi41MTdjMCw0LjI2My0zLjQ1Myw3LjcxNi03LjcxNiw3LjcxNmgtMzIuODM3ICBjLTQuMjYzLDAtNy43MTYtMy40NTUtNy43MTYtNy43MTZWMjI0Ljg1NmMwLTQuMjYzLDMuNDU1LTcuNzE2LDcuNzE2LTcuNzE2aDMyLjgzOWM0LjI2MywwLDcuNzE2LDMuNDU1LDcuNzE2LDcuNzE2djEzNy42NjEgIEgyMDAuMTU1eiBNMTc2LjAyLDIwNC4xNjJjLTE3LjIzLDAtMzEuMTk2LTEzLjk2OS0zMS4xOTYtMzEuMjAxYzAtMTcuMjI5LDEzLjk2NS0zMS4xOTYsMzEuMTk2LTMxLjE5NiAgYzE3LjIzMiwwLDMxLjE5OCwxMy45NjksMzEuMTk4LDMxLjE5NkMyMDcuMjE4LDE5MC4xOTUsMTkzLjI1MSwyMDQuMTYyLDE3Ni4wMiwyMDQuMTYyTDE3Ni4wMiwyMDQuMTYyeiBNMzcxLjc3MywzNjMuMTM5ICBjMCwzLjkxOC0zLjE3NSw3LjA5Ni03LjA5LDcuMDk2aC0zNS4yNDJjLTMuOTE1LDAtNy4wOTItMy4xNzctNy4wOTItNy4wOTZ2LTY0LjU3NGMwLTkuNjMsMi44MjYtNDIuMjEtMjUuMTcyLTQyLjIxICBjLTIxLjcxOSwwLTI2LjEyNCwyMi4zLTI3LjAwOCwzMi4zMDR2NzQuNDc4YzAsMy45MTgtMy4xNzUsNy4wOTYtNy4wOTQsNy4wOTZoLTM0LjA4MmMtMy45MTgsMC03LjA5Ni0zLjE3Ny03LjA5Ni03LjA5NlYyMjQuMjM0ICBjMC0zLjkxOCwzLjE3Ny03LjA5Niw3LjA5Ni03LjA5NmgzNC4wODJjMy45MTgsMCw3LjA5NCwzLjE3Nyw3LjA5NCw3LjA5NnYxMi4wMDljOC4wNTQtMTIuMDg1LDIwLjAyMy0yMS40MTYsNDUuNTAyLTIxLjQxNiAgYzU2LjQyNSwwLDU2LjEwNSw1Mi43MTcsNTYuMTA1LDgxLjY4NnY2Ni42MjdIMzcxLjc3M3oiLz4KPHBhdGggc3R5bGU9ImZpbGw6I0QxRDFEMTsiIGQ9Ik0zNzYuMDIzLDEwOS40NjhIMjU1LjQyN3YxMDcuNjdoNy42NDdjMy45MTgsMCw3LjA5NCwzLjE3Nyw3LjA5NCw3LjA5NnYxMi4wMDkgIGM4LjA1NC0xMi4wODUsMjAuMDIzLTIxLjQxNiw0NS41MDMtMjEuNDE2YzU2LjQyNSwwLDU2LjEwNSw1Mi43MTcsNTYuMTA1LDgxLjY4NnY2Ni42MjVjMCwzLjkxOC0zLjE3NSw3LjA5Ni03LjA5LDcuMDk2aC0zNS4yNDIgIGMtMy45MTUsMC03LjA5Mi0zLjE3Ny03LjA5Mi03LjA5NnYtNjQuNTc0YzAtOS42MywyLjgyNi00Mi4yMS0yNS4xNzItNDIuMjFjLTIxLjcxOSwwLTI2LjEyNCwyMi4zLTI3LjAwOCwzMi4zMDR2NzQuNDc4ICBjMCwzLjkxOC0zLjE3NSw3LjA5Ni03LjA5NCw3LjA5NmgtNy42NDd2MzIuMzAxaDEyMC41OTZjMTQuNjM5LDAsMjYuNTEtMTEuODcxLDI2LjUxLTI2LjUxVjEzNS45NzggIEM0MDIuNTMzLDEyMS4zMzUsMzkwLjY2MiwxMDkuNDY4LDM3Ni4wMjMsMTA5LjQ2OHoiLz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPC9zdmc+Cg==);
        }
    }
}
