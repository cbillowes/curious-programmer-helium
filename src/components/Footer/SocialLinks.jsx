import React, { Component } from "react"
import ExternalLink from "../Link/ExternalLink"
import data from "../../../data/SocialLinks"
import "./SocialLinks.scss"

class License extends Component {
  render() {
    const links = data.footer

    return (
      <ul className="social-links-footer">
        {links.map(item => {
          return (
            <li key={`social-links-footer-${item.icon}`} className="link">
              <ExternalLink
                to={item.href}
                className={item.icon} />
            </li>)
        })}
      </ul>
    )
  }
}

export default License
