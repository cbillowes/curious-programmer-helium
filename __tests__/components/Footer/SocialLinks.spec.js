import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import SocialLinks from "../../../src/components/Footer/SocialLinks"
import data from "../../../data/SocialLinks"

describe("a footer social links component", () => {
  let element = <SocialLinks />
  let component
  
  beforeAll(() => {
    component = mount(element)
  })

  it("should render", () => {
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render container", () => {
    expect(component.find(".social-links-footer").length).toEqual(1)
  })

  it("should render copyright content", () => {
    data.footer.map(item => {
      expect(component.html()).toContain(item.name.toLowerCase())
      expect(component.html()).toContain(`href="${item.href}"`)
    })
  })
})