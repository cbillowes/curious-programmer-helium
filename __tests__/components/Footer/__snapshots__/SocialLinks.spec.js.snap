// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a footer social links component should render 1`] = `
<ul
  className="social-links-footer"
>
  <li
    className="link"
  >
    <a
      className="github"
      href="https://github.com/cbillowes"
      rel="nofollow noopener noreferrer "
      target="_blank"
    />
  </li>
  <li
    className="link"
  >
    <a
      className="twitter"
      href="https://twitter.com/cbillowes"
      rel="nofollow noopener noreferrer "
      target="_blank"
    />
  </li>
  <li
    className="link"
  >
    <a
      className="linkedin"
      href="https://www.linkedin.com/in/cbouwer/"
      rel="nofollow noopener noreferrer "
      target="_blank"
    />
  </li>
</ul>
`;
